src/mbgl/map/transform.cpp

```diff
@@ -517,7 +527,7 @@ double Transform::getAngle() const {
 #pragma mark - Pitch
 
 void Transform::setPitch(double pitch, const Duration& duration) {
-    setPitch(pitch, {NAN, NAN}, duration);
+    setPitch(pitch, ScreenCoordinate {}, duration);
```

Comment:
If we don't specify a valid anchor (by _valid_ I mean an anchor that does not contain any NaN and contains at least one non-zero value on its axis),  these are simply not taken in consideration in `startTransition()`.